lms/djangoapps/instructor/views/data_access.py

```diff
@@ -106,6 +115,8 @@ def get_group_query_students(course_id, group_id):
     _group, queries, _relation = get_saved_queries(course_id, group_id)
     # wait for the queries to finish before proceeding
     make_subqueries.apply_async(args=(course_id, group_id, queries)).get()
+    # the previous functions alter the state so we call flush_transactions to make sure we are most up-to-date
+    flush_transactions()
```

Comment:
we're committing before we read. as far as i understand it, we open a transaction in the beginning of the function. when we make a call to another function in the middle, it opens a transaction there (so we won't see its changes until we call flush in this one)